hphp/util/hash-crc-arm.S

```diff
@@ -0,0 +1,189 @@
+#include "hphp/util/etch-helpers.h"
+#include "hphp/util/hphp-config.h"
+
+#if defined(ENABLE_AARCH64_CRC) && !defined(NO_HWCRC) && \
+    !defined(__CYGWIN__) && !defined(__MINGW__) && !defined(_MSC_VER)
+
+.file   "hphp/util/hash-crc-arm.S"
+
+#define HASH_FUNC_NAME _ZN4HPHP20hash_string_i_unsafeEPKcj
+
+ETCH_SECTION(HASH_FUNC_NAME)
+.globl    HASH_FUNC_NAME
+ETCH_TYPE(HASH_FUNC_NAME, @function)
+ETCH_ALIGN16
+ETCH_NAME(HASH_FUNC_NAME):
+        CFI(startproc)
+        negs    w1, w1
+        mov     w9, #0xffffffff
+        beq     ETCH_LABEL(iend)
+        mov     w3, w1
+        mov     x4, #0xdfdfdfdfdfdfdfdf
+        b       ETCH_LABEL(iheader)
+
+ETCH_ALIGN16
```

Comment:
We should have the alignment there.  That is the inner loop should be aligned so the start of the loop is aligned for the fetch units of the common processors.  The ones Cavium has loads 16 byte aligned from the data cache IIRC.  So this is a good thing for our processors.    Now I noticed this is a straight copy of the x86 code which might not be a good thing.